librz/core/cmd/cmd_regs.c

```diff
@@ -0,0 +1,603 @@
+	if (item->size < 80) {
+		ut64 value = rz_reg_get_value(reg, item);
+		snprintf(out, out_size, "0x%08" PFMT64x, value);
+	} else {
+		utX valueBig;
+		rz_reg_get_value_big(reg, item, &valueBig);
+		switch (item->size) {
+		case 80:
+			snprintf(out, out_size, "0x%04x%016" PFMT64x "", valueBig.v80.High, valueBig.v80.Low);
+			break;
+		case 96:
+			snprintf(out, out_size, "0x%08x%016" PFMT64x "", valueBig.v96.High, valueBig.v96.Low);
+			break;
+		case 128:
+			snprintf(out, out_size, "0x%016" PFMT64x "%016" PFMT64x "", valueBig.v128.High, valueBig.v128.Low);
+			break;
+		case 256:
+			snprintf(out, out_size, "0x%016" PFMT64x "%016" PFMT64x "%016" PFMT64x "%016" PFMT64x "",
+				valueBig.v256.High.High, valueBig.v256.High.Low, valueBig.v256.Low.High, valueBig.v256.Low.Low);
+			break;
+		default:
+			snprintf(out, out_size, "ERROR");
+		}
+	}
+}
+
+/// Check whether the given item's value has changed in the last step
+static bool reg_has_changed(RzReg *reg, RzRegItem *item) {
+	ut64 value = rz_reg_get_value(reg, item);
+	rz_reg_arena_swap(reg, false);
+	ut64 old = rz_reg_get_value(reg, item);
+	rz_reg_arena_swap(reg, false);
+	return old != value;
+}
+
+static void print_reg_not_found(const char *arg) {
+	RZ_LOG_ERROR("No such register or register type: \"%s\"\n", rz_str_get(arg));
+}
+
+/**
+ * \brief (Sub)handler for register assignments like reg=0x42
+ * \param arg the full argument string, like "reg = 0x42"
+ * \param eq_pos index of the '=' in arg
+ */
+static RzCmdStatus assign_reg(RzCore *core, RzReg *reg, const char *arg, size_t eq_pos) {
+	char *str = strdup(arg);
+	if (!str) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	str[eq_pos] = 0;
+	char *val = str + eq_pos + 1;
+	rz_str_trim(str);
+	rz_str_trim(val);
+	RzRegItem *ri = rz_reg_get(reg, str, RZ_REG_TYPE_ANY);
+	if (!ri) {
+		free(str);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	ut64 nval = rz_num_math(core->num, val);
+	rz_reg_set_value(reg, ri, nval);
+	return RZ_CMD_STATUS_OK;
+}
+
+static const char *get_reg_color(RzCore *core, RzReg *reg, RzRegItem *item) {
+	if (!rz_config_get_i(core->config, "scr.color")) {
+		return NULL;
+	}
+	if (reg_has_changed(reg, item)) {
+		return (core->cons && core->cons->context->pal.creg)
+			? core->cons->context->pal.creg
+			: Color_BWHITE;
+	}
+	return NULL;
+}
+
+static const char *get_reg_role_name(RzReg *reg, RzRegItem *item) {
+	for (int i = 0; i < RZ_REG_NAME_LAST; i++) {
+		const char *t = rz_reg_get_name(reg, i);
+		if (t && !strcmp(t, item->name)) {
+			return rz_reg_get_role(i);
+		}
+	}
+	return NULL;
+}
+
+static RzCmdStatus show_regs_handler(RzCore *core, RzReg *reg, const char *filter, RzCmdStateOutput *state) {
+	RzList *ritems = filter_reg_items(reg, filter);
+	if (!ritems) {
+		print_reg_not_found(filter);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	switch (state->mode) {
+	case RZ_OUTPUT_MODE_STANDARD: {
+		RzListIter *iter;
+		RzRegItem *item;
+		char buf[256] = { 0 };
+		rz_list_foreach (ritems, iter, item) {
```

Comment:
What I tend to do in my handlers is to have the `rz_list_foreach` outside of the switch because it is something that needs to be done in all modes, so it feels just easier to have common code.